docs/communicate-with-javascript.md

```diff
@@ -1,3822 +1,39 @@
-
-See the [Calling an object method](#calling-an-object-method) section for more
-information.
-
-#### `arr.sort(compareFunction)`: mutating instance method
-
-<!--#prelude#let arr = [|2|]-->
-```ocaml
-external sort : 'a array -> (('a -> 'a -> int)[@mel.uncurry]) -> 'a array
-  = "sort"
-[@@mel.send]
-
-let _ = sort arr compare
-```
-```reasonml
-[@mel.send]
-external sort: (array('a), [@mel.uncurry] (('a, 'a) => int)) => array('a) =
-  "sort";
-
-let _ = sort(arr, compare);
-```
-
-For a mutating method, it's traditional to pass the instance argument first.
-
-Note: `compare` is a function provided by the standard library, which fits the
-defined interface of JavaScript's comparator function.
-
-### Null and undefined
-
-#### `foo.bar === undefined`: check for undefined
+features to create *bindings* to JavaScript code. A binding is a piece of code
+that references a JavaScript object or value and makes it available for use in
+Melange code. A very basic example of a binding is:
 
 ```ocaml
-module Foo = struct
-  type t
-  external bar : t -> int option = "bar" [@@mel.get]
-end
-
-external foo : Foo.t = "foo"
-
-let _result = match Foo.bar foo with Some _ -> 1 | None -> 0
+external confirm : string -> bool = "window.confirm"
 ```
 ```reasonml
-module Foo = {
-  type t;
-  [@mel.get] external bar: t => option(int) = "bar";
-};
-
-external foo: Foo.t = "foo";
-
-let _result =
-  switch (Foo.bar(foo)) {
-  | Some(_) => 1
-  | None => 0
-  };
-```
-
-If you know some value may be `undefined` (but not `null`, see next section),
-and if you know its type is monomorphic (i.e. not generic), then you can model
-it directly as an `Option.t` type.
-
-See the [Non-shared data types](#non-shared-data-types) section for more
-information.
-
-#### `foo.bar == null`: check for null or undefined
-
-```ocaml
-module Foo = struct
-  type t
-  external bar : t -> t option = "bar" [@@mel.get] [@@mel.return nullable]
-end
-
-external foo : Foo.t = "foo"
-
-let _result = match Foo.bar foo with Some _ -> 1 | None -> 0
+external confirm: string => bool = "window.confirm";
 ```
-```reasonml
-module Foo = {
-  type t;
-  [@mel.get] [@mel.return nullable] external bar: t => option(t) = "bar";
-};
 
-external foo: Foo.t = "foo";
-
-let _result =
-  switch (Foo.bar(foo)) {
-  | Some(_) => 1
-  | None => 0
-  };
-```
+The above code binds JavaScript's
+[window.confirm](https://developer.mozilla.org/en-US/docs/Web/API/Window/confirm)
+function to a Melange function named `confirm`. Note how the binding includes
+type annotations so that there's no ambiguity about the types of the arguments
+and return value.
 
-If you know the value is 'nullable' (i.e. could be `null` or `undefined`), or if
-the value could be polymorphic, then `mel.return nullable` is appropriate to
-use.
+---
 
-Note that this attribute requires the return type of the binding to be an
-`option` type as well.
+Because the topic of bindings is quite big, we've broken it up into multiple
```

Comment:
Why mention the breaking?

Reply:
Oops, that's a mistake. Fixed in https://github.com/melange-re/melange-re.github.io/pull/208/commits/96b04514c2b8c5c7d6a3a4e6188b05b6c16242e8